File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        l = 0
        res = cur = blocks[:k].count('W')
        for r in range(k,len(blocks)):
            if blocks[r] == 'W':
                cur += 1
            if blocks[l] == 'W':
                cur -= 1
                res = min(res, cur)
            l += 1
        return res

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int l = 0;
        int cur = count(blocks.begin(), blocks.begin()+k, 'W');
        int res = cur;
        for (int r=k; r<blocks.size(); r++) {
            if (blocks[r] == 'W') cur++;
            if (blocks[l] == 'W') {
                cur--;
                res = min(res, cur);
            }
            l++;
        }
        return res;
    }
};